### PR TITLE
Feature/post put delete admin api

### PR DESCRIPTION
**Pull request by CowboyBebop**
User Admin and Project Admin APIs

### CODE REVIEW

File: src/routes/api/admin/users.ts [hunk 1]
```diff
@@ -28,16 +27,53 @@ adminUsersRouter.get("/:userId", async (req, res, next) => {
...
+  try {
+    //Find projects from DB
+    const userDoc = await UserModel.findById(req.params.userId);
+
+    if (!userDoc) return next(createError(404, "user not found"));
+
+    userDoc.banned = false;
+    await userDoc.save();
+
+    return res.json(userDoc);
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/users.ts]: Just return a 200, no need to return the doc.

File: src/routes/api/admin/users.ts [hunk 2]
```diff
@@ -28,16 +27,53 @@ adminUsersRouter.get("/:userId", async (req, res, next) => {
...
+  try {
+    //Find projects from DB
+    const userDoc = await UserModel.findById(req.params.userId);
+
+    if (!userDoc) return next(createError(404, "user not found"));
+
+    userDoc.banned = true;
+    await userDoc.save();
+
+    return res.json(userDoc);
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/users.ts]: Just return a 200, no need to return the doc.

Comment (Pierre-Demessence) [reply src/routes/api/admin/users.ts]: Also should fail (400) if user already banned.

File: src/routes/api/admin/users.ts (hunk 1 above)
Comment (Pierre-Demessence) [reply src/routes/api/admin/users.ts]: Also should fail (400) if user already not banned.

File: src/routes/api/admin/users.ts [hunk 3]
```diff
@@ -28,16 +27,53 @@ adminUsersRouter.get("/:userId", async (req, res, next) => {
...
+//TODO: Admin Authenticate
+adminUsersRouter.delete("/:userId", async (req, res, next) => {
+  try {
+    //Find projects from DB
+    const userDoc = await UserModel.findById(req.params.userId);
+
+    if (!userDoc) return next(createError(404, "user not found"));
+    userDoc.deleteOne();
+
+    return res.json(userDoc);
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/users.ts]: Just return a 200, no need to return the doc.

File: src/routes/api/admin/projects.ts [hunk 4]
```diff
@@ -28,8 +28,19 @@ adminProjectsRouter.get("/:projectId", async (req, res, next) => {
...
+//TODO: Admin Authenticate
+adminProjectsRouter.delete("/:projectId", async (req, res, next) => {
+  try {
+    //Find projects from DB
+    const projectDoc = await ProjectModel.findById(req.params.projectId);
+
+    if (!projectDoc) return next(createError(404, "project not found"));
+    projectDoc.deleteOne();
+
+    return res.json(projectDoc);
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/projects.ts]: Just return a 200, no need to return the doc.

File: src/routes/api/admin/projects.ts [hunk 5]
```diff
@@ -28,8 +28,19 @@ adminProjectsRouter.get("/:projectId", async (req, res, next) => {
...
-adminProjectsRouter.delete("/:projectId", (req, res, next) => {
-  res.send("deleted a project");
+//TODO: Admin Authenticate
+adminProjectsRouter.delete("/:projectId", async (req, res, next) => {
+  try {
+    //Find projects from DB
+    const projectDoc = await ProjectModel.findById(req.params.projectId);
+
+    if (!projectDoc) return next(createError(404, "project not found"));
+    projectDoc.deleteOne();
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/projects.ts]: Better to await, so the response is sent only AFTER the doc is deleted.

File: src/routes/api/admin/users.ts [hunk 6]
```diff
@@ -28,16 +27,53 @@ adminUsersRouter.get("/:userId", async (req, res, next) => {
...
-adminUsersRouter.delete("/:userId", (req, res, next) => {
-  res.send("deleted a user");
+//TODO: Admin Authenticate
+adminUsersRouter.delete("/:userId", async (req, res, next) => {
+  try {
+    //Find projects from DB
+    const userDoc = await UserModel.findById(req.params.userId);
+
+    if (!userDoc) return next(createError(404, "user not found"));
+    userDoc.deleteOne();
```
Comment (Pierre-Demessence) [inline src/routes/api/admin/users.ts]: Better to await, so the response is sent only AFTER the doc is deleted.

File: src/server.ts [hunk 7]
```diff
@@ -12,6 +12,7 @@ import routes from "./routes/indexRoutes";
 import "./mongodb/db";
 
 process.on("uncaughtException", async (err) => {
+  console.log(err);
```
Comment (Pierre-Demessence) [inline src/server.ts:15]: ??

File: src/util/authUtil.ts [hunk 8]
```diff
@@ -5,5 +5,7 @@ export async function generateSalt(): Promise<string> {
 }
 
 export async function hashPassword(password: string, passwordSalt: string): Promise<string> {
+  console.log(password);
```
Comment (Pierre-Demessence) [inline src/util/authUtil.ts]: Please don't push console logs

**Review by Pierre-Demessence — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 7 above)
Comment (CowboyBebop) [reply src/server.ts:15]: The console error shows where the error is, differently from the error stuck only. At least that's what my console showed me.

**Pull request merged**